### PR TITLE
Manage Learners: fixed User.post_save handler causing migrations to fail

### PR DESCRIPTION
**Pull request by e-kolpakov**
**Description:** Fixes migration/DB deployment error caused by User.post_save handler.

**Related:** #8

**JIRA:** [ENT-48](https://openedx.atlassian.net/browse/ENT-48)

**Testing instructions:**

* Run bokchoy test suite - should get past "Loading the fixture data from the filesystem into the default MySQL DB." and actually run the suite. ([failed bokchoy run](https://build.testeng.edx.org/job/edx-platform-accessibility-pr/27918/console) for reference).
* Migrate `commerce` and `enterprise` to zero (you might need to fake `commerce` 0001 and manually delete auth_user record)
* Apply commerce 0001 migration only. *Expected result:* migration works. *Failure condition:* migration fails complaining about missing enterprise_pendingenterprisecustomeruser table.
* Unapply commerce 0001 (might need faking again).
* Migrate everything at once (`./manage.py lms migrate --settings=devstack). **Expected result:* migrations succeed.

**Reviewers:**
- [x] @mtyaka
- [x] @asadiqbal08 and or @saleem-latif 
- [ ] (Optional) @mattdrayer 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [x] Commits are squashed
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by e-kolpakov** [comment]
@mtyaka, @saleem-latif, @asadiqbal08 Another PR to fix DB-related issues.

File: enterprise/models.py [hunk 1]
```diff
@@ -336,6 +335,13 @@ def handle_user_post_save(sender, **kwargs):  # pylint: disable=unused-argument
         pending_link_record = PendingEnterpriseCustomerUser.objects.get(user_email=user_instance.email)
     except PendingEnterpriseCustomerUser.DoesNotExist:
         return  # nothing to do in this case
+    except (ProgrammingError, OperationalError):
+        logger.exception(
+            "DB-related error - skipping the rest of the handler. "
```
Comment (mattdrayer) [inline enterprise/models.py]: @e-kolpakov -- can you summarize what the issue was that has been addressed by this PR?  I guess I'm not following why the `PendingEnterpriseCustomerUser` model/table would not be available.  Perhaps we could include a bit more info in this exception message, besides "Hey don't worry about this error, it's all good."

**Comment by e-kolpakov** [comment]
@mtyaka @asadiqbal08 @saleem-latif please review

File: tests/test_signals.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                                  PendingEnterpriseCustomerUserFactory)
+
+
+@mark.django_db
+@ddt.ddt
+class TestUserPostSaveSignalHandler(unittest.TestCase):
+    """
+    Test User post_save signal handler.
+
+    This class does not user UserFactory intentionally
```
Comment (mtyaka) [inline tests/test_signals.py]: nit/typo: "user" -> "use"; missing dot at end of sentence.

**Comment by e-kolpakov** [comment]
@mtyaka @saleem-latif Thank you for the review guys, but there's another problem - current version works fine with bokchoy (because users are bulk loaded), but fails to provision the DB from scratch, as there's one migration in `commerce` app that adds User normally. So, in addition to what we have here now, it should disable the handler during migration phase or be tolerant to DB errors when querying `PendingEnterpriseCustomerUser`. The latter approach is simpler, but at the risk of suppressing valid exceptions. So I'll make at least one update here today.

**Comment by e-kolpakov** [comment]
@mtyaka @saleem-latif Patch is sent - now it should work properly during migration phase - please take another look (I've submitted it as a separate commit for your convenience and will squash after review).

**Comment by e-kolpakov** [comment]
@mtyaka @saleem-latif FYI - I've changed a couple of things on this PR, so technically it invalidates your previous approvals - please take another look (and note we've piled up a couple of things on top of it already, so the sooner you can do that the better). Sorry for this back-and-forth.

**Comment by e-kolpakov** [comment]
@mtyaka @saleem-latif I don't think we can wait any longer, so I'm merging this - any notes will be addressed in follow up PRs.
